.github/workflows/acctest.yml

```diff
@@ -71,7 +71,7 @@ jobs:
         version:
           - stable
         terraform:
-          - '1.1.5'
+          - '1.5'
```

Comment:
The `terraform_data` resource was introduced in 1.4, so we need to run tests in 1.4+ in order to use that resource.  I _think_ this might need to go in a separate PR so that it can be merged to `main` separately because we use the `pull_request_target` trigger.

Reply:
Do you want to merge a PR with that change first so we can see these tests pass?

Reply:
I kicked off a manual run of the updated job, so we can wait to see what that does: https://github.com/equinix/terraform-provider-equinix/actions/runs/6411225404/job/17406201790

Reply:
Since the tests are flaky even without the caching introduced here, I re-ran them a couple times to build confidence that these changes are not causing new test failures.  I don't see anything to indicate that this change is bringing in _new_ problems, and it does appear to stabilize the tests in terms of avoiding unnecessary plan changes due to changes in plans API responses due to platform capacity.